great_expectations/datasource/data_connector/inferred_asset_gcs_data_connector.py

```diff
@@ -1,7 +1,162 @@
+            batch_spec_passthrough (dict): dictionary with keys that will be added directly to batch_spec
+        """
+        logger.debug(f'Constructing InferredAssetGCSDataConnector "{name}".')
+
+        super().__init__(
+            name=name,
+            datasource_name=datasource_name,
+            execution_engine=execution_engine,
+            default_regex=default_regex,
+            sorters=sorters,
+            batch_spec_passthrough=batch_spec_passthrough,
+        )
+
+        self._bucket_or_name = bucket_or_name
+        self._prefix = prefix
+        self._delimiter = delimiter
+        self._max_results = max_results
+
+        if gcs_options is None:
+            gcs_options = {}
+
+        try:
+            credentials = None  # If configured with gcloud CLI / env vars
+            if "filename" in gcs_options:
+                filename = gcs_options.pop("filename")
+                credentials = service_account.Credentials.from_service_account_file(
+                    filename=filename
+                )
+            elif "info" in gcs_options:
+                info = gcs_options.pop("info")
+                credentials = service_account.Credentials.from_service_account_info(
+                    info=info
+                )
+            self._gcs = storage.Client(credentials=credentials, **gcs_options)
+        except (TypeError, AttributeError):
+            raise ImportError(
+                "Unable to load GCS Client (it is required for InferredAssetGCSDataConnector)."
+            )
+
+    def build_batch_spec(self, batch_definition: BatchDefinition) -> GCSBatchSpec:
+        """
+        Build BatchSpec from batch_definition by calling DataConnector's build_batch_spec function.
+
+        Args:
+            batch_definition (BatchDefinition): to be used to build batch_spec
+
+        Returns:
+            BatchSpec built from batch_definition
+        """
+        batch_spec: PathBatchSpec = super().build_batch_spec(
+            batch_definition=batch_definition
+        )
+        return GCSBatchSpec(batch_spec)
+
+    def _get_data_reference_list(
+        self, data_asset_name: Optional[str] = None
+    ) -> List[str]:
+        query_options: dict = {
+            "bucket_or_name": self._bucket_or_name,
+            "prefix": self._prefix,
+            "delimiter": self._delimiter,
+            "max_results": self._max_results,
+        }
+
+        path_list: List[str] = [
+            key
+            for key in list_gcs_keys(
+                gcs=self._gcs,
+                query_options=query_options,
+                recursive=True,
+            )
+        ]
+        return path_list
+
+    def _get_full_file_path(
+        self,
+        path: str,
+        data_asset_name: Optional[str] = None,
+    ) -> str:
+        # data_asset_name isn't used in this method.
+        # It's only kept for compatibility with parent methods.
+        return f"gs://{os.path.join(self._bucket_or_name, path)}"
```

Comment:
Just to confirm, would this change at all if the behavior of `_bucket_or_name` were to change in a future PR?

Reply:
Nope just the naming would change! The one change to note is that the query options dict passed to list_gcs_keys would need to reassign the value from "bucket" to "bucket_or_name" to work.

Would not impact the end user at all and is a simple naming change :)